dali/test/python/reader/test_numpy.py

```diff
@@ -192,6 +192,42 @@ def _get_type_and_shape_params():
                         num_threads, batch_size
 
 
+@params(
+    (False, ),
+    (True, ),
+)
+def test_header_parse(use_o_direct):
+    # Test different ndims to see how well we handle headers of different lengths and padding.
+    # The NPY token (header meta-data) + header is padded to the size aligned up to 64 bytes.
+    # In particular the `np.full((1,) * 21, 1., dtype=float32)` and
+    # `np.full((1,) * 22, 1., dtype=float32)` are the boundary between 128 and 192 bytes header.
+    # This make a good case for testing the bounds we use for extracting the header.
+    # The 32 is the max dimensionality handled by the numpy
+    ndims = list(range(33))
+    with tempfile.TemporaryDirectory(prefix=gds_data_root) as test_data_root:
```

Comment:
Do we really need to test _all_ of them? Wouldn't a small subset like `1, 2, 3, 7, 16, 32` do?

Reply:
The most interesting case is 21, as that has minimal padding after the header, which makes it sensitive to small offset errors. But the test is very simple and lightweight so I went for all the cases.